garando_syntax/src/util/move_map.rs

```diff
@@ -29,13 +29,13 @@ impl<T> MoveMap<T> for Vec<T> {
             while read_i < old_len {
                 // move the read_i'th item out of the vector and map it
                 // to an iterator
-                let e = ptr::read(self.get_unchecked(read_i));
+                let e = ptr::read(self.as_mut_ptr().offset(read_i as isize));
```

Comment:
I did not comment these casts but I'm a noob to this sort of manipulation in Rust.

My reading of https://doc.rust-lang.org/reference/type-layout.html#primitive-data-layout, though, particularly "usize and isize have a size big enough to contain every address on the target platform.", seems to suggest this is always OK.